core/src/client/async_client/mod.rs

```diff
@@ -403,163 +620,46 @@ async fn background_task<S, R>(
 		let next_backend = backend_event.next();
 		futures_util::pin_mut!(next_frontend, next_backend);
 
-		match futures_util::future::select(next_frontend, next_backend).await {
-			// User dropped the sender side of the channel.
-			// There is nothing to do just terminate.
-			Either::Left((None, _)) => {
-				tracing::trace!("[backend]: frontend dropped; terminate client");
-				break;
-			}
-
-			Either::Left((Some(FrontToBack::Batch(batch)), _)) => {
-				tracing::trace!("[backend]: client prepares to send batch request: {:?}", batch.raw);
-				// NOTE(niklasad1): annoying allocation.
-				if let Err(send_back) = manager.insert_pending_batch(batch.ids.clone(), batch.send_back) {
-					tracing::warn!("[backend]: batch request: {:?} already pending", batch.ids);
-					let _ = send_back.send(Err(Error::InvalidRequestId));
-					continue;
-				}
-
-				if let Err(e) = sender.send(batch.raw).await {
-					tracing::warn!("[backend]: client batch request failed: {:?}", e);
-					manager.complete_pending_batch(batch.ids);
-				}
-			}
-			// User called `notification` on the front-end
-			Either::Left((Some(FrontToBack::Notification(notif)), _)) => {
-				tracing::trace!("[backend]: client prepares to send notification: {:?}", notif);
-				if let Err(e) = sender.send(notif).await {
-					tracing::warn!("[backend]: client notif failed: {:?}", e);
-				}
-			}
-
-			// User called `request` on the front-end
-			Either::Left((Some(FrontToBack::Request(request)), _)) => {
-				tracing::trace!("[backend]: client prepares to send request={:?}", request);
-				match sender.send(request.raw).await {
-					Ok(_) => manager
-						.insert_pending_call(request.id, request.send_back)
-						.expect("ID unused checked above; qed"),
-					Err(e) => {
-						tracing::warn!("[backend]: client request failed: {:?}", e);
-						let _ = request.send_back.map(|s| s.send(Err(Error::Transport(e.into()))));
-					}
-				}
-			}
+		let mut submit_ping = if let Some(duration) = ping_interval {
+			Delay::new(duration).fuse()
+		} else {
+			Fuse::<Delay>::terminated()
```

Comment:
Perhaps worth a comment like "Polling this will always return pending"? (it's not super obvious from the name to me!)